src/python/tests/core/bot/fuzzers/afl/afl_engine_test.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for AFL's engine implementation."""
+
+import os
+import shutil
+import unittest
+
+from bot.fuzzers.afl import engine
+from bot.fuzzers.afl import launcher
+from system import environment
+from tests.core.bot.fuzzers.afl import afl_launcher_integration_test
+from tests.test_libs import helpers as test_helpers
+
+# TODO(mbarbella): Break dependency on afl_launcher_integration_test once
+# everything has been fully converted to the new pipeline.
+
+TEST_PATH = os.path.abspath(os.path.dirname(__file__))
+TEMP_DIRECTORY = os.path.join(TEST_PATH, 'temp')
+DATA_DIRECTORY = os.path.join(TEST_PATH, 'data')
+CORPUS_DIRECTORY = os.path.join(TEMP_DIRECTORY, 'corpus')
+OUTPUT_DIRECTORY = os.path.join(TEMP_DIRECTORY, 'output')
+
+FUZZ_TIMEOUT = (5 + launcher.AflRunnerCommon.SIGTERM_WAIT_TIME +
+                launcher.AflRunnerCommon.AFL_CLEAN_EXIT_TIME)
+
+
+def clear_temp_dir():
+  """Clear temp directories."""
+  if os.path.exists(TEMP_DIRECTORY):
+    shutil.rmtree(TEMP_DIRECTORY)
+
+
+def create_temp_dir():
+  """Create temp directories."""
+  # Corpus directory will be created when preparing for fuzzing.
+  os.mkdir(TEMP_DIRECTORY)
+  os.mkdir(OUTPUT_DIRECTORY)
+
+
+@unittest.skipIf(not environment.get_value('AFL_INTEGRATION_TESTS'),
+                 'AFL_INTEGRATION_TESTS=1 must be set')
+class AFLEngineTest(unittest.TestCase):
+  """Tests for AFLEngine."""
+
+  def setUp(self):
+    clear_temp_dir()
+    create_temp_dir()
+
+    test_helpers.patch_environ(self)
+    afl_launcher_integration_test.dont_use_strategies(self)
+
+  def tearDown(self):
+    clear_temp_dir()
+
+  def test_fuzz(self):
+    """Test for fuzz."""
+    engine_impl = engine.AFLEngine()
+
+    _ = afl_launcher_integration_test.setup_testcase_and_corpus(
```

Comment:
nit: Don't even assign if we aren't using the result.

Reply:
Done!